config-example.yaml

```diff
@@ -224,54 +224,63 @@ policy:
 # - https://tailscale.com/kb/1081/magicdns/
 # - https://tailscale.com/blog/2021-09-private-dns-with-magicdns/
 #
-dns_config:
-  # Whether to prefer using Headscale provided DNS or use local.
-  override_local_dns: true
+# Please not that for the DNS configuration to have any effect,
+# `macic_dns` must be enabled.
+# In addition, clients that are to use the DNS configuration
+# must have the `--accept-dns` option enabled.
+#
+# Enabling `magic_dns` and `--accept-dns` on the clients will
+# integrate with the DNS manager on the client or overwrite
+# /etc/resolv.conf.
+# https://tailscale.com/kb/1235/resolv-conf
+dns:
+  # Whether to use [MagicDNS](https://tailscale.com/kb/1081/magicdns/).
+  # Only works if there is at least a nameserver defined.
+  magic_dns: true
+
+  # Defines the base domain to create the hostnames for MagicDNS.
+  # `base_domain` must be a FQDNs, without the trailing dot.
+  # The FQDN of the hosts will be
+  # `hostname.base_domain` (e.g., _myhost.example.com_).
+  base_domain: example.com
 
   # List of DNS servers to expose to clients.
   nameservers:
-    - 1.1.1.1
-
-  # NextDNS (see https://tailscale.com/kb/1218/nextdns/).
-  # "abc123" is example NextDNS ID, replace with yours.
-  #
-  # With metadata sharing:
-  # nameservers:
-  #   - https://dns.nextdns.io/abc123
-  #
-  # Without metadata sharing:
-  # nameservers:
-  #   - 2a07:a8c0::ab:c123
-  #   - 2a07:a8c1::ab:c123
-
-  # Split DNS (see https://tailscale.com/kb/1054/dns/),
-  # list of search domains and the DNS to query for each one.
-  #
-  # restricted_nameservers:
-  #   foo.bar.com:
-  #     - 1.1.1.1
-  #   darp.headscale.net:
-  #     - 1.1.1.1
-  #     - 8.8.8.8
-
-  # Search domains to inject.
-  domains: []
+    global:
+      - 1.1.1.1
+      - 1.0.0.1
+      - 2606:4700:4700::1111
+      - 2606:4700:4700::1001
```

Comment:
For this cloudflare example it looks like they have a DoH endpoint at: `https://cloudflare-dns.com/dns-query` (from https://developers.cloudflare.com/1.1.1.1/encryption/dns-over-https/make-api-requests/)

Reply:
I would think that it would work, but I have not actually tested it, happy to include it if someone can test it, but that can be a follow up PR too.

Reply:
Another question: by default resolv.conf only takes 3 name servers. How does tailscale handle more?

Reply:
resolv.conf is a "fallback" for tailscale, [it will try a number of ways before using resolv.conf](https://tailscale.com/blog/sisyphean-dns-client-linux).

In this case, when you override DNS, the DNS server in resolv.conf will be 100.100.100.100, which lives inside the tailscale daemon, and it will proxy everything.

Reply:
you can see this in the integration tests https://github.com/juanfont/headscale/pull/2034/files#diff-6534f1d1f61ff53e7fe7de90f7736900614802c5bdaa10a63f485316b95351c2R126